src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -461,10 +461,23 @@ private void parseDoc(Map<String, String> namespacePrefixesByUri) {
                 final String instanceId = instanceNodeIdStrs.get(instanceIndex);
                 final String instanceSrc = instance.getAttributeValue(null, "src");
 
-                if (instanceSrc != null && instanceSrc.toLowerCase().startsWith("jr://file")) { // file or file-csv
+                final String externalSrc;
```

Comment:
This change is super helpful to understand what's going on. Thanks!

I'd suggest you extract it to a method, so that we can have a nice method name that improves semantics and externalSrc can be immutable in the context of `parseDoc()`